spring-pulsar/src/test/java/org/springframework/pulsar/listener/PulsarListenerTests.java

```diff
@@ -955,4 +957,53 @@ void pulsarMessagesBatchListenerWithHeaders(Messages<String> data,
 
 	}
 
+	@Nested
+	@ContextConfiguration(classes = ConsumerPauseTest.ConsumerPauseConfig.class)
+	class ConsumerPauseTest {
+
+		private static final CountDownLatch latch = new CountDownLatch(10);
+
+		@Autowired
+		PulsarListenerEndpointRegistry pulsarListenerEndpointRegistry;
+
+		@Test
+		void containerPauseAndResumeSuccessfully() throws Exception {
```

Comment:
Great test! It may be helpful to add a test that continues to put data on the topic while the container is paused (another thread) with a large `receiverQueueSize` in order to ensure it does what we expect in that case.